libs/server/Resp/Objects/ObjectStoreUtils.cs

```diff
@@ -65,5 +66,40 @@ private bool AbortWithErrorMessage(int count, ReadOnlySpan<byte> errorMessage)
 
             return true;
         }
+
+        /// <summary>
+        /// Tries to parse the input as "LEFT" or "RIGHT" and returns the corresponding OperationDirection.
+        /// If parsing fails, returns OperationDirection.Unknown.
+        /// </summary>
+        /// <param name="input">The input to parse.</param>
+        /// <returns>The parsed OperationDirection, or OperationDirection.Unknown if parsing fails.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public OperationDirection GetOperationDirection(ReadOnlySpan<byte> input)
```

Comment:
Use this version to optimize for the common case and avoid strings for pre-NET8"

```cs
        public OperationDirection GetOperationDirection(ArgSlice input)
        {
            // Optimize for the common case
            if (input.ReadOnlySpan.SequenceEqual("LEFT"u8))
                return OperationDirection.Left;
            if (input.ReadOnlySpan.SequenceEqual("RIGHT"u8))
                return OperationDirection.Right;
            // Rare case: try making upper case and retry
            MakeUpperCase(input.ptr);
            if (input.ReadOnlySpan.SequenceEqual("LEFT"u8))
                return OperationDirection.Left;
            if (input.ReadOnlySpan.SequenceEqual("RIGHT"u8))
                return OperationDirection.Right;
            return OperationDirection.Unknown;
        }
```

Reply:
Great, you've taken into account the majority of cases and the minority of cases!